### ISSUE TITLE
"Error: Cannot find module 'knex.js'" from running test suite

### ISSUE THREAD

**Issue by joshkel** [knex/knex#3935]
# Environment

Knex version: master (b59161f)
Database + version: N/A
OS: Ubuntu 20.04 with Node 12.18.2 via NodeSource

# Bug

I'm trying to clone the Knex repository and run the test suite and am getting errors within the test suite.

For example:

```sh
mkdir -p ~/tmp
cd ~/tmp
git clone git@github.com:knex/knex.git
cd knex
npm i
```

If I then manually run one of the tests that I'm interested in:

```sh
node_modules/.bin/mocha --exit -t 10000 test/all-tests-suite.js --grep 'ignores migrationSource if migration directory is specified'
```

I get the following error:

```
 0 passing (284ms)
  1 failing

  1) CLI tests
       knexfile
         knexfile supports custom migrationSource
           ignores migrationSource if migration directory is specified:
     node /home/josh/tmp/knex/bin/cli.js migrate:latest --knexfile=test/jake-util/knexfile-custom-migration-source/knexfile-with-directory.js --knexpath=../knex.js -> FAIL. 
Stdout: Working directory changed to ~/tmp/knex/test/jake-util/knexfile-custom-migration-source
 
Error: Error: Cannot find module 'knex.js'
Require stack:
- /home/josh/tmp/knex/bin/cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)
    at Function.Module._load (internal/modules/cjs/loader.js:842:27)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at initKnex (/home/josh/tmp/knex/bin/cli.js:57:16)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Command.<anonymous> (/home/josh/tmp/knex/bin/cli.js:168:26)
```

(I'm not sure if running individual tests like this is intended to be supported. I got similar errors from running the full test suite - but I had other problems mixed in with the larger test suite, so I was trying to focus on this particular error.)

If I edit bin/cli.js to get the full path to knex.js, it works:

```js
function invoke(env) {
  env.modulePath = path.resolve(env.modulePath || env.knexpath || process.env.KNEX_PATH);
```

However, I don't understand Knex's cwd and module design well enough to know if this fix is correct or not. I'm happy to open a PR if it would help.

Thank you.

**Comment by elhigu** [comment]
Running individual tests is mostly not supported (many of the tests are dependent on each other).

**Comment by joshkel** [comment]
Sorry for the delayed reply.

I was probably unclear in my original issue; I got numerous errors from trying to run the whole test suite, and so I started attempting to run individual failing tests, which is where I ran into this problem.

With that said - I don't currently have a need to do development on Knex (as evidenced by how long it took me to reply to this ☹️ ), so feel free to close this issue if it's not worth further attention right now.
